htmlPlugin.js

```diff
@@ -0,0 +1,40 @@
+const { appInfo: { loki } } = require('./package.json');
+
+const resourceRootUrn = `urn:com:${loki.cloudCodeName}:${loki.appCodeName}:app:pages:${loki.pageCodeName}!`;
+/**
+ * @param {string} htmlstring An HTML string
+ * @param {string} filename A bundle file name to replace with "${loki.web.resourceUrl('[b]')}""
+ */
+function replacePathWithLokiResource(htmlstring, filename) {
+  const regexp = new RegExp(
+    `(=")(?:)[^=]*(?:)(${filename})(?:)[^=]*(")`,
+    'g',
+  );
+  return htmlstring.replace(
+    regexp,
+    `$1\${loki.web.resourceUrl('${resourceRootUrn}$2')}$3`,
+  );
+}
+const htmlPlugin = () => {
+  /** @type {import('vite').Plugin} */
+  const def = {
+    name: 'html-transform',
+    enforce: 'post',
+    transformIndexHtml(html, ctx) {
```

Comment:
Where possible, try to avoid abbreviations in variable names and function names. It makes things more verbose, but I err on the side of verbosity vs nomenclature that's easy to lose track of.

Reply:
Okay I won't do that.

Reply:
So this one isn't **_my_** fault! [Vite Specific Hooks](https://vitejs.dev/guide/api-plugin.html#transformindexhtml).